components/style/gecko_bindings/sugar/ns_css_value.rs

```diff
@@ -173,6 +173,25 @@ impl nsCSSValue {
     pub fn set_from<T: ToNsCssValue>(&mut self, value: &T) {
         value.convert(self)
     }
+
+    /// Returns an `Angle` value from this `nsCSSValue`.
+    ///
+    /// Panics if the unit is not `eCSSUnit_Degree` `eCSSUnit_Grad`, `eCSSUnit_Turn`
+    /// or `eCSSUnit_Radian`.
+    pub fn get_angle(&self) -> Angle {
+        unsafe {
+            Angle::from_gecko_values(self.float_unchecked(), self.mUnit)
+        }
+    }
+
+    /// Sets Angle value to this nsCSSValue.
+    pub fn set_angle(&mut self, angle: Angle) {
+        let (value, unit) = angle.to_gecko_values();
+        self.mUnit = unit;
```

Comment:
Oh, one thing I missed, please assert here we're not leaking anything (that our unit isn't one of the `heap` ones).